### PR TITLE
Enhance Profile Management and UI Tests on feature/update-bottom-bar Branch

### PR DESCRIPTION
**Pull request by aoutz**
**PR Update:** This PR continues development on the feature/update-bottom-bar branch to avoid creating new branches and follows up on my previous PR.

This update includes:

- Merged latest changes from main.
- Added UI tests for SeekerProfileScreen and EditSeekerProfileScreen to ensure UI stability and correct functionality.
- Adapted SeekerProfileScreen and EditSeekerProfileScreen to include test tags for easier testing.
- Applied ktfmt for consistent formatting. 

This PR builds on previous work #48 done with my colleague rimabk for profile management #18 , #17 . Key features include ProfileViewModel, ProfileScreen, EditProfileScreen, and Firebase integration for user profile data management and persistence.

### CODE REVIEW

File: app/src/main/java/com/android/solvit/seeker/ui/profile/SeekerProfile.kt [hunk 1]
```diff
@@ -276,14 +54,13 @@ fun SeekerProfileScreen(viewModel: SeekerProfileViewModel, navigationActions: Na
...
+        modifier = Modifier.testTag("profileScreen"),
         topBar = {
           TopAppBar(
-              modifier = Modifier.testTag("ProfileTopBar"),
               backgroundColor = Color.Blue, // Set blue background
               title = { Text("Profile", color = Color.White) },
               navigationIcon = {
-                // When you go back from profile you diretctly return to home page
-                IconButton(onClick = { navigationActions.navigateTo(Route.SERVICES) }) {
+                IconButton(onClick = { navigationActions.goBack() }) {
```
Comment (hjamaly) [inline app/src/main/java/com/android/solvit/seeker/ui/profile/SeekerProfile.kt:63]: I think that we don't to do a goBack but more a navigation to the HomePage for a smoother user experience

**Review by hjamaly — APPROVED** [review]
This PR introduces significant improvements, especially with the new tests ensuring that the SeekerProfileScreen and EditSeekerProfileScreen function correctly—great job on implementing these! The tests add valuable stability to the profile management feature and help reinforce UI reliability.

However, there is a user experience issue I'd like to highlight. Currently, the goBack action on the profile management screen returns the user to the previous screen rather than the homepage. It might be more intuitive and user-friendly if the goBack action redirected directly to the homepage, streamlining navigation and reducing unnecessary steps. Would this adjustment align with the intended flow?

File: app/src/main/java/com/android/solvit/seeker/ui/profile/SeekerProfile.kt
Comment (achillet3) [inline app/src/main/java/com/android/solvit/seeker/ui/profile/SeekerProfile.kt:1]: Nit: I agree with Hassan on the goBack behavior.

**Review by achillet3 — APPROVED** [review]
# Overall comment
This PR adds value to the project by introducing tests for both `EditSeekerProfile.kt` and `SeekerProfile.kt` screens. Very good line coverage well done !

# Potential Improvements
While testing the UI good practice is to have an `@After` function to unmock everything and be sure to start from scratch (with your correctly implemented @Before function). This would help when testing the conditional coverage in the future. However, since you test all the UI components' presence in a single test this is not necessary.

**Pull request closed without merging**